tests/testthat/test-shinytest2-tm_g_ci.R

```diff
@@ -0,0 +1,231 @@
+        select = teal.transform::select_spec(
+          choices = c("ARMCD", "BMRKR2"),
+          selected = c("ARMCD"),
+          multiple = FALSE,
+          fixed = FALSE
+        )
+      ),
+      y_var = teal.transform::data_extract_spec(
+        dataname = "ADLB",
+        filter = list(
+          teal.transform::filter_spec(
+            vars = "PARAMCD",
+            choices = levels(ADLB$PARAMCD),
+            selected = levels(ADLB$PARAMCD)[1],
+            multiple = FALSE,
+            label = "Select lab:"
+          ),
+          teal.transform::filter_spec(
+            vars = "AVISIT",
+            choices = levels(ADLB$AVISIT),
+            selected = levels(ADLB$AVISIT)[1],
+            multiple = FALSE,
+            label = "Select visit:"
+          )
+        ),
+        select = teal.transform::select_spec(
+          label = "Analyzed Value",
+          choices = c("AVAL", "CHG"),
+          selected = "AVAL",
+          multiple = FALSE,
+          fixed = FALSE
+        )
+      ),
+      color = teal.transform::data_extract_spec(
+        dataname = "ADSL",
+        select = teal.transform::select_spec(
+          label = "Color by variable",
+          choices = c("SEX", "STRATA1", "STRATA2"),
+          selected = c("STRATA1"),
+          multiple = FALSE,
+          fixed = FALSE
+        )
+      )
+    )
+  )
+}
+
+testthat::test_that("e2e - tm_g_ci: example ci module initializes in teal without errors and produces plot output", {
+  skip_if_too_deep(5)
+  app_driver <- app_driver_tm_g_ci()
+  app_driver$expect_no_shiny_error()
+  app_driver$expect_no_validation_error()
+  testthat::expect_match(active_module_pws_output(app_driver), "data:image/png;base64,")
+})
+
+testthat::test_that("e2e - tm_g_ci: Module initializes with specified label, x_var, y_var, ADLB filters and color", {
+  skip_if_too_deep(5)
+  app_driver <- app_driver_tm_g_ci()
+
+  testthat::expect_equal(
+    app_driver$get_text("#teal-main_ui-root-active_tab > li.active > a"),
+    "Confidence Interval Plot"
+  )
+
+  testthat::expect_equal(
+    app_driver$get_active_module_input("x_var-dataset_ADSL_singleextract-select"),
+    "ARMCD"
+  )
+
+  testthat::expect_equal(
+    app_driver$get_active_module_input("y_var-dataset_ADLB_singleextract-select"),
+    "AVAL"
+  )
+
+  testthat::expect_equal(
+    app_driver$get_active_module_input("y_var-dataset_ADLB_singleextract-filter1-col"),
+    "PARAMCD"
+  )
+  testthat::expect_equal(
+    app_driver$get_active_module_input("y_var-dataset_ADLB_singleextract-filter1-vals"),
+    "ALT"
+  )
+
+  testthat::expect_equal(
+    app_driver$get_active_module_input("y_var-dataset_ADLB_singleextract-filter2-col"),
+    "AVISIT"
+  )
+  testthat::expect_equal(
+    app_driver$get_active_module_input("y_var-dataset_ADLB_singleextract-filter2-vals"),
+    "SCREENING"
+  )
+
+  testthat::expect_equal(
+    app_driver$get_active_module_input("color-dataset_ADSL_singleextract-select"),
+    "STRATA1"
+  )
```

Comment:
this is minor, as you are testing those 2 inputs later, but we can also verify their default values
```suggestion
  )
  testthat::expect_equal(
    app_driver$get_active_module_input("conf_level"),
    "0.95"
  )
  
  testthat::expect_equal(
    app_driver$get_active_module_input("stat"),
    "mean"
  )
```